code/modules/mob/living/carbon/human/physiology.dm

```diff
@@ -25,5 +25,75 @@
 
 	var/do_after_speed = 1 //Speed mod for do_after. Lower is better. If temporarily adjusting, please only modify using *= and /=, so you don't interrupt other calculations.
 
+	///Static ssoc list of levels (ints) - strings
+	var/list/static/level_names = list("Novice", "Apprentice", "Journeyman", "Expert", "Master", "Legendary")//This list is already in the right order, due to indexing
+	///Assoc list of skills - level
+	var/list/skills = list()
+	///Assoc list of skills - exp
+	var/list/skill_experience = list()
+
 /datum/physiology/New()
 	armor = new
+
+///Adjust experience of a specific skill
+/datum/physiology/proc/adjust_experience(skill, amt, user)
+	skill_experience[skill] = max(0, skill_experience[skill] + amt) //Prevent going below 0
+	var/old_level = skills[skill]
+	switch(skill_experience[skill])
+		if(SKILL_EXP_LEGENDARY to INFINITY)
+			skills[skill] = SKILL_LEVEL_LEGENDARY
+		if(SKILL_EXP_MASTER to SKILL_EXP_LEGENDARY)
+			skills[skill] = SKILL_LEVEL_MASTER
+		if(SKILL_EXP_EXPERT to SKILL_EXP_MASTER)
+			skills[skill] = SKILL_LEVEL_EXPERT
+		if(SKILL_EXP_JOURNEYMAN to SKILL_EXP_EXPERT)
+			skills[skill] = SKILL_LEVEL_JOURNEYMAN
+		if(SKILL_EXP_APPRENTICE to SKILL_EXP_JOURNEYMAN)
+			skills[skill] = SKILL_LEVEL_APPRENTICE
+		if(SKILL_EXP_NOVICE to SKILL_EXP_APPRENTICE)
+			skills[skill] = SKILL_LEVEL_NOVICE
+		if(0 to SKILL_EXP_NOVICE)
+			skills[skill] = SKILL_LEVEL_NONE	
+	if(skills[skill] == old_level)
+		return //same level
+	if(skills[skill] >= old_level)
+		to_chat(user, "<span class='nicegreen'>I feel like I've become more proficient at [skill]!</span>")
```

Comment:
`My Skill level is rounded out and full.`

Reply:
@Qustinnus eating miners brains should transfer their XP to you